Backend/api/views/transaction_view.py

```diff
@@ -32,28 +33,35 @@ def post(self, request):
 
                 utxo = utxo + tran['value']
                 pub_list = list(address.redeem_script.split(' '))
+                
                 pub_list.pop(0)
-                pub_list.pop(len(pub_list) - 1)
-                pub_list.pop(len(pub_list) - 1)
+                while len(pub_list) > 3:
+                    pub_list.pop(len(pub_list) - 1)
+                
                 redeem=RedeemScript.create_p2sh_multisig(quorum_m=2,pubkey_hexes=pub_list,expected_addr_network='testnet')
+                print(redeem)
                 
                 inp=TxIn(bytes.fromhex(tran['txid']) , tran['vout'], script_sig=redeem)
                 txn_inputs.append(inp)
 
+        print(utxo)
+        fee = (len(txn_inputs) + 2) * 146 + 2 * 34 + 20
+        print(fee)
         output1= TxOut.to_address(recipient, amount_to_send)
-        output2= TxOut.to_address(address.address_generated, utxo - amount_to_send + 1000)
+        output2= TxOut.to_address(address.address_generated, utxo - (amount_to_send + fee))
 
         txn=Tx(2, txn_inputs, [output1, output2], network='testnet')
-        
 
-        
         if len(request.data['private_keys']) == 2:
             p_k_WIF_1 = request.data['private_keys'][0]
             p_k_WIF_2 = request.data['private_keys'][1]
         else:
+            p_k_WIF_1 = request.data['private_keys'][0]
             p_k_WIF_2 = address.service_key
 
         try:
+            print(p_k_WIF_1)
+            print(p_k_WIF_2)
             pkey1=PrivateKey.parse(p_k_WIF_1)
             pkey2=PrivateKey.parse(p_k_WIF_2)
         except:
```

Comment:
what should happen after exception

Reply:
Nothing, there is a response that would be returned

Reply:
okay, I did not see it though